lib/src/Uncertainty/Algorithm/Simulation/CrossEntropyImportanceSamplingAlgorithm.cxx

```diff
@@ -0,0 +1,245 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief CrossEntropyImportanceSamplingAlgorithm implement parent class for Cross Entropy Importance Sampling algorithms
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/CrossEntropyImportanceSamplingAlgorithm.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class CrossEntropyImportanceSamplingAlgorithm
+ */
+
+CLASSNAMEINIT(CrossEntropyImportanceSamplingAlgorithm)
+
+static const Factory<CrossEntropyImportanceSamplingAlgorithm> Factory_CrossEntropyImportanceSamplingAlgorithm;
+
+// Default constructor
+CrossEntropyImportanceSamplingAlgorithm::CrossEntropyImportanceSamplingAlgorithm()
+  : EventSimulation()
+{
+  // Nothing TO DO
+}
+
+
+// Default constructor
+CrossEntropyImportanceSamplingAlgorithm::CrossEntropyImportanceSamplingAlgorithm(const RandomVector & event,
+                                                                                 const Scalar rhoQuantile)
+  : EventSimulation(event)
+  , initialDistribution_(event.getAntecedent().getDistribution())
+{
+  event_ = event;
```

Comment:
event_ is already set via the parent class

Reply:
fixed